src/preloader/TcpSocketElectron.ts

```diff
@@ -115,57 +164,58 @@ export class TcpSocketElectron {
     // where sockets have file descriptors. See
     // <https://github.com/nodejs/help/issues/1312>
     // eslint-disable-next-line no-underscore-dangle
-    return (this._socket as unknown as MaybeHasFd)._handle?.fd;
+    return (this.#socket as unknown as MaybeHasFd)._handle?.fd;
   }
 
-  // eslint-disable-next-line @foxglove/no-boolean-parameters
+  // eslint-disable-next-line @lichtblick/no-boolean-parameters
   setKeepAlive(enable?: boolean, initialDelay?: number): this {
-    this._socket.setKeepAlive(enable, initialDelay);
+    this.#socket.setKeepAlive(enable, initialDelay);
     return this;
   }
 
   setTimeout(timeout: number): this {
-    this._socket.setTimeout(timeout);
+    this.#socket.setTimeout(timeout);
     return this;
   }
 
-  // eslint-disable-next-line @foxglove/no-boolean-parameters
+  // eslint-disable-next-line @lichtblick/no-boolean-parameters
   setNoDelay(noDelay?: boolean): this {
-    this._socket.setNoDelay(noDelay);
+    this.#socket.setNoDelay(noDelay);
     return this;
   }
 
   connected(): boolean {
-    return !this._socket.destroyed && this._socket.localAddress != undefined;
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    return !this.#socket.destroyed && this.#socket.localAddress != undefined;
   }
 
   async connect(): Promise<void> {
-    return await new Promise((resolve, reject) => {
-      this._socket
+    await new Promise<void>((resolve, reject) => {
+      this.#socket
         .connect({ host: this.host, port: this.port, lookup: dnsLookup }, () => {
-          this._socket.removeListener("error", reject);
+          this.#socket.removeListener("error", reject);
           resolve();
-          this._emit("connect");
+          this.#emit("connect");
         })
         .on("error", reject);
     });
   }
 
   close(): void {
-    this._socket.destroy();
+    this.#socket.destroy();
   }
 
   dispose(): void {
-    this._socket.removeAllListeners();
+    this.#socket.removeAllListeners();
     this.close();
-    this._messagePort.close();
+    this.#messagePort.close();
   }
```

Comment:
Change return type to string and returns a message like "Connection closed"

```suggestion
  dispose(): string {
    this.#socket.removeAllListeners();
    this.close();
    this.#messagePort.close();
    return "Connection closed";
  }
```

Reply:
Changed, thanks for the suggestion